pkg/crc/machine/machine.go

```diff
@@ -204,6 +205,21 @@ func Start(startConfig StartConfig) (StartResult, error) {
 		return *result, err
 	}
 
+	// kubeconfig file is also needed inside the VM to perform pull secret replace operation.
+	kubeConfigFile, err := ioutil.ReadFile(filepath.Join(constants.MachineInstanceDir, machineConfig.Name, "kubeconfig"))
+	if err != nil {
+		logging.ErrorF("Failed to read %s: %v", filepath.Join(constants.MachineInstanceDir, machineConfig.Name, "kubeconfig"), err)
+		result.Error = err.Error()
+		return *result, err
+	}
+	//	Update the user pull secret before kubelet start.
+	logging.Info("Adding user's pull secret ...")
```

Comment:
We only need to do this on first start, it seems you are doing it on every start?

Reply:
yes, if a user do stop/start that is not required but atm we are just forcefully doing for every start.

Reply:
How long does this take approximately when you tested this?

Reply:
@cfergeau ~30-40 sec.

Reply:
This is quite a long time to wait for each `crc start`, I think it's important to only do it when it's needed.